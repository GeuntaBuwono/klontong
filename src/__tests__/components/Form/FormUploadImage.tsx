import FormUploadImage from '@components/Form/FormUploadImage';
import {render, screen} from '@testing-library/react';

describe('FormUploadImage', () => {
	const testElement = (
		<FormUploadImage
			placeholder="placeholder testing"
			id="uploadImage"
			onBlur={jest.fn()}
			onChange={jest.fn()}
		/>
	);

	const testElementWithError = (
		<FormUploadImage
			placeholder="placeholder testing"
			errorMessage="has an error"
			id="uploadImage"
			onBlur={jest.fn()}
			onChange={jest.fn()}
		/>
	);

	test('snapshot', () => {
		expect(testElement).toMatchSnapshot();
	});

	test('render correctly', () => {
		render(testElement);
		const testInput = screen.getByLabelText('placeholder testing');
		expect(testInput).toBeInTheDocument();
	});

	test('has an error', () => {
		render(testElementWithError);
		const testInput = screen.getByLabelText('placeholder testing');
		expect(testInput).toBeTruthy();
	});
});
