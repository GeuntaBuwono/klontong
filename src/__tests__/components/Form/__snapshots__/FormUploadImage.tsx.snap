// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormUploadImage snapshot 1`] = `
<FormUploadImage
  id="uploadImage"
  onBlur={[MockFunction]}
  onChange={[MockFunction]}
  placeholder="placeholder testing"
/>
`;
